Sources/BuildSystemIntegration/ExternalBuildSystemAdapter.swift

```diff
@@ -178,6 +185,48 @@ actor ExternalBuildSystemAdapter {
     ).connection
   }
 
+  private func getConfigPath() -> AbsolutePath? {
+    var buildServerConfigLocations: [AbsolutePath?] = []
+
+    let workspacePath = projectRoot.appending(component: ".bsp")
+    buildServerConfigLocations.append(workspacePath)
+
+    #if os(Windows)
+    let localAppDataPath = AbsolutePath(validatingOrNil: ProcessInfo.processInfo.environment["LOCALAPPDATA"])
+    let programDataPath = AbsolutePath(validatingOrNil: ProcessInfo.processInfo.environment["PROGRAMDATA"])
+
+    buildServerConfigLocations.append(contentsOf: [localAppDataPath, programDataPath])
+    #else
+    let xdgDataHomePath = AbsolutePath(validatingOrNil: ProcessInfo.processInfo.environment["XDG_DATA_HOME"])
+    let xdgDataDirsPath = AbsolutePath(validatingOrNil: ProcessInfo.processInfo.environment["XDG_DATA_DIRS"])
+
+    buildServerConfigLocations.append(contentsOf: [xdgDataHomePath, xdgDataDirsPath])
+
+    if let libraryUrl = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first {
+      let libraryPath = AbsolutePath(validatingOrNil: libraryUrl.absoluteString)
+      buildServerConfigLocations.append(libraryPath)
+    }
```

Comment:
We should keep the ordering here. `$XDG_DATA_HOME/bsp/` should be before `$HOME/Library/Application Support/bsp/` and that before `$XDG_DATA_DIRS/bsp/`, just in case somebody has the `XDG` variables set on macOS.

Reply:
That's exactly how it's ordered, I have appended `XDG` paths before the `library` paths. So if it finds in either of them it's returned first.

Reply:
No, it’s not. According to the spec, if you have a build server in `$HOME/Library/Application Support/bsp/` and also in `$XDG_DATA_DIRS/bsp/`, we should pick the one in `$HOME/Library/Application Support/bsp/`. But with the current implementation, we pick the one in `$XDG_DATA_DIRS/bsp/`.

Reply:
> We should keep the ordering here. $XDG_DATA_HOME/bsp/ should be before $HOME/Library/Application Support/bsp/

I thought you meant to prefer `XDG` before `library` when you say "before"? I think I misunderstood.